programs/bpf_loader/src/lib.rs

```diff
@@ -1936,28 +1943,32 @@ mod tests {
             is_writable: false,
         };
 
+        let loaded_program = Arc::new(load_test_program(&program_account, loader_id));
```

Comment:
I think it would be better to do the pre-loading of the BPF programs in `with_mock_invoke_context!()` automatically instead of doing it in all tests manually.

Reply:
Wouldn't that require changes to lot more tests that actually do not need to load the programs? Since the signature of the macro will change to pass list of programs that'll be loaded.